src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Presentation/SampleChooserViewModel.cs

```diff
@@ -1266,13 +1266,15 @@ private async Task DumpOutputFolderName(CancellationToken ct, string folderName)
 					border.Background = _screenshotBackground;
 				}
 
-				element.InvalidateMeasure();
-				element.InvalidateArrange();
-				await Task.Yield();
+				// This won't do anything on WinUI since the layout will immediately be reapplied the moment we let go
+				// of the Task and it will undo these measure/arrange changes, so we should make sure to take the
+				// snapshot before the runtime has a chance to relayout. Since RenderAsync is, well, async, we can't
+				// guarantee that. We match this behaviour of relayouting on wasm and skia, but it works anyway due to
+				// the way we implement synchronization and RenderAsync, which let us grab the snapshot before the
+				// relayouting.
 				element.Measure(new Windows.Foundation.Size(constraints.Width, constraints.Height));
```

Comment:
Tested on WinUI only, this doesn't work as expected both before and after the change.

IMO, This measure call should not be needed. IMO, https://github.com/unoplatform/uno/pull/15825 could be better. It also makes sure we screenshot the whole window, and avoids these unnecessary measures.